src/System.Runtime.Numerics/src/System/Numerics/Complex.cs

```diff
@@ -162,7 +166,9 @@ private static double Hypot(double a, double b)
         private static double Log1P(double x)
         {
             // Compute log(1 + x) without loss of accuracy when x is small.
+
             // Our only use case so far is for positive values, so this isn't coded to handle negative values.
+            Debug.Assert(!(x < 0.0));
```

Comment:
Ah, yes (it is so easy to let NaN slip your mind when reading code). In either case, I definitely think explicitly calling out `double.IsNaN` is useful here and will help prevent future refactorings from breaking things.